canonical-web-and-design/jaas-dashboard
# Execute action when selecting an action and hitting submit

## Done

- When selecting an action and populating any options then clicking submit, it'll send that action request to Juju.

TODO: Tests

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- View a charm with actions then click on the actions link (contact me to have one shared)
- Open your devtools and clear the network tab to make it easier to see the requests.
- Interaction 1
- Click an action without options, the submit button should enable
- Click submit
- You should see the action request being made.
- Interaction 2
- Click an action with options, then fill out any required fields, the submit button should enable
- Click submit
- You should see the action request being made.

## Details

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1664
